### PR TITLE
Add JsonPatchDocument

### LINKED ISSUE #13474: Prototype a JSON Patch document type.
Spec: http://jsonpatch.com/

### PR DESCRIPTION
**Pull request by pakrym**
I've simplified a type a little after Krzysztofs initial feedback.

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/13474

### CODE REVIEW

File: sdk/core/Azure.Core.Experimental/src/JsonPatchDocument.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        public void AppendReplace(string path, string rawJsonValue)
+        {
+            Operations.Add(new JsonPatchOperation(JsonPatchOperationKind.Replace, path, null, rawJsonValue));
+        }
+
+        /// <summary>
+        /// Appends a "copy" operation to this <see cref="JsonPatchDocument"/>.
+        /// </summary>
+        /// <param name="from">The path to copy from.</param>
+        /// <param name="path">The path to copy to.</param>
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.Experimental/src/JsonPatchDocument.cs:51]: Should this be named "to" or "fromPath"/"toPath"

File: sdk/core/Azure.Core.Experimental/src/JsonPatchOperationKind.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma warning disable 1591
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.Experimental/src/JsonPatchOperationKind.cs:4]: Hmm why was this needed?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/core/Azure.Core.Experimental/src/JsonPatchDocument.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core.Experimental/src/JsonPatchDocument.cs:51]: I'm trying to keep the names similar between parameter and the resulting json.

File: sdk/core/Azure.Core.Experimental/src/JsonPatchOperationKind.cs (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core.Experimental/src/JsonPatchOperationKind.cs:4]: At one point types were public and I was lazy adding comments. Removed.

**Review by drwill-ms — APPROVED** [review]
